tests/fake/gitlab.go

```diff
@@ -15,7 +17,10 @@ type GitLab struct {
 	port int
 	Echo *echo.Echo
 
+	client *http.Client
+
 	nextWebhookID int
+	projectHooks  map[string]*gitlab.WebhookPost
```

Comment:
I feel it's still better to separate project and project hook.

Also we may report error if we create the same project/hook multiple times.

Reply:
I think create repository creates a webhook in a project. And one project may contain multiple webhooks, so I change the mapping to a list. This kinda follows gitlab behavior.